engine/src/main/java/org/terasology/i18n/TranslationSystem.java

```diff
@@ -37,18 +24,21 @@
 
     /**
      * If no perfect match is found for the default locale, fallback strategies will attempt to find the closest match.
-     * @param id must match the pattern <code>${module:project#id}</code>
+     * @param id must match the pattern {@code ${module:project#id}}
+     * @param arguments {@link java.text.MessageFormat} arguments for the translation string
      * @return the translated string
      */
-    String translate(String id);
+    String translate(String id, Object... arguments);
```

Comment:
The NUI PR #4017 touches this file, but it does not remove it. I think the changes should be compatible without major disruption here.